doc/source/tune/examples/optuna_example.ipynb

```diff
@@ -0,0 +1,329 @@
+    "    return {\"steps\": 100}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca86258a",
+   "metadata": {},
+   "source": [
+    "Optionally, we may provide an initial set of hyperparameters that we believe as the current best, (`current_best_params`), and\n",
+    "pass this belief into a `OptunaSearch` searcher.\n",
+    "Furthermore, we optionally define a `scheduler` to go along with our algorithm.\n",
+    "\n",
+    "We then set the maximum concurrent trials to `4` with a `ConcurrencyLimiter`.\n",
+    "Lastly, we must set the number of samples for this Tune run to `1000`\n",
+    "(you can decrease this if it takes too long on your machine)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53b9a2d2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "initial_params = [\n",
+    "    {\"width\": 1, \"height\": 2, \"activation\": \"relu\"},\n",
+    "    {\"width\": 4, \"height\": 2, \"activation\": \"relu\"},\n",
+    "]\n",
+    "\n",
+    "# Choosing specified case:\n",
+    "if not multi_obj: \n",
+    "    if not use_func:\n",
+    "        # Initial guess\n",
+    "        searcher = OptunaSearch(points_to_evaluate=initial_params)\n",
+    "    else:\n",
+    "        # Search space from function, initial guess\n",
+    "        searcher = OptunaSearch(space=define_by_run_func, points_to_evaluate=initial_params)\n",
+    "else: \n",
+    "    if not use_func:\n",
+    "        # Multi-objective, initial guess\n",
+    "        searcher = OptunaSearch(points_to_evaluate=initial_params, metric=[\"loss\", \"gain\"], mode=[\"min\", \"max\"])\n",
+    "    else:\n",
+    "        # Multi-objective, search space from function, initial guess\n",
+    "        searcher = OptunaSearch(space=define_by_run_func, points_to_evaluate=initial_params, metric=[\"loss\", \"gain\"], mode=[\"min\", \"max\"])\n",
+    "\n",
+    "algo = ConcurrencyLimiter(searcher, max_concurrent=4)\n",
+    "scheduler = AsyncHyperBandScheduler()\n",
+    "\n",
+    "num_samples = 1000"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1616059e",
+   "metadata": {
+    "lines_to_next_cell": 0,
+    "tags": [
+     "remove-cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# If 1000 samples take too long, you can reduce this number.\n",
+    "# We override this number here for our smoke tests.\n",
+    "num_samples = 10"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bcd4d81d",
+   "metadata": {},
+   "source": [
+    "Finally, all that's left is running the experiment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fd506370",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "analysis = tune.run(\n",
+    "    objective if not multi_obj else multi_objective,\n",
+    "    search_alg=algo,\n",
+    "    scheduler=scheduler if not multi_obj else None,\n",
+    "    metric=\"mean_loss\" if not multi_obj else None,\n",
```

Comment:
@thetwotravelers can we please just make this two separate cells? Or delete the case in which it's not multi-objective (which we've seen already above).

Reply:
went with two separate cells. is that ok? i realize all the `if` and `else` business is a bit ugly to look at, if there is another way to combine examples without such a structure, please let me know. would it be actually be preferable to simply stack the examples into a very long notebook? i assumed that's even less ideal.